framework/src/org/checkerframework/framework/util/typeinference/TypeArgInferenceUtil.java

```diff
@@ -188,9 +197,17 @@ public static AnnotatedTypeMirror assignedTo(AnnotatedTypeFactory atypeFactory,
             AnnotatedExecutableType constructor = AnnotatedTypes.asMemberOf(types, atypeFactory, receiver, constructorElt);
             int treeIndex = -1;
             for (int i = 0; i < constructor.getParameterTypes().size(); ++i) {
-                if (TreeUtils.skipParens(newClassTree.getArguments().get(i)) == path.getLeaf()) {
+                Tree argumentTree = TreeUtils.skipParens(newClassTree.getArguments().get(i));
+                if (argumentTree == path.getLeaf()) {
                     treeIndex = i;
                     break;
+                } else if (argumentTree.getKind() == Kind.CONDITIONAL_EXPRESSION) {
+                    ConditionalExpressionTree conditionalExpressionTree = (ConditionalExpressionTree) argumentTree;
```

Comment:
What about nested conditionals and parens within conditionals?
Can you extend the test with `test1( foo ? ( baz ? x :  ((y))) : (new Issue672))` or some such?